x-pack/plugins/stack_connectors/server/connector_types/bedrock/bedrock.ts

```diff
@@ -135,23 +193,46 @@ export class BedrockConnector extends SubActionConnector<Config, Secrets> {
     messages,
     model,
   }: InvokeAIActionParams): Promise<InvokeAIActionResponse> {
-    const combinedMessages = messages.reduce((acc: string, message) => {
-      const { role, content } = message;
-      // Bedrock only has Assistant and Human, so 'system' and 'user' will be converted to Human
-      const bedrockRole = role === 'assistant' ? '\n\nAssistant:' : '\n\nHuman:';
-      return `${acc}${bedrockRole}${content}`;
-    }, '');
-
-    const req = {
-      // end prompt in "Assistant:" to avoid the model starting its message with "Assistant:"
-      prompt: `${combinedMessages} \n\nAssistant:`,
-      max_tokens_to_sample: DEFAULT_TOKEN_LIMIT,
-      temperature: 0.5,
-      // prevent model from talking to itself
-      stop_sequences: ['\n\nHuman:'],
-    };
-
-    const res = await this.runApi({ body: JSON.stringify(req), model });
+    const res = await this.runApi({ body: JSON.stringify(formatBedrockBody({ messages })), model });
     return { message: res.completion.trim() };
   }
 }
+
+const formatBedrockBody = ({
+  messages,
+}: {
+  messages: Array<{ role: string; content: string }>;
+}) => {
+  const combinedMessages = messages.reduce((acc: string, message) => {
+    const { role, content } = message;
+    // Bedrock only has Assistant and Human, so 'system' and 'user' will be converted to Human
+    const bedrockRole = role === 'assistant' ? '\n\nAssistant:' : '\n\nHuman:';
+    return `${acc}${bedrockRole}${content}`;
+  }, '');
+
+  return {
+    // end prompt in "Assistant:" to avoid the model starting its message with "Assistant:"
+    prompt: `${combinedMessages} \n\nAssistant:`,
+    max_tokens_to_sample: DEFAULT_TOKEN_LIMIT,
+    temperature: 0.5,
+    // prevent model from talking to itself
+    stop_sequences: ['\n\nHuman:'],
+  };
+};
+
+const transformToSimpleString = () =>
+  new Transform({
+    transform(chunk, encoding, callback) {
+      const encoder = new TextEncoder();
+      const codec = new EventStreamCodec(toUtf8, fromUtf8);
+      const event = codec.decode(chunk);
+      const body = JSON.parse(
+        Buffer.from(
+          JSON.parse(new TextDecoder('utf-8').decode(event.body)).bytes,
+          'base64'
+        ).toString()
+      );
```

Comment:
I could not find documentation on how to handle their codec. I did find https://github.com/awsdocs/aws-doc-sdk-examples/issues/5541 for when documentation should be added.

I followed the bedrock extension in the langchainjs repo here: https://github.com/langchain-ai/langchainjs/pull/2367/files#diff-ead6be41360b3bf1c4559ea4d0069a015b69a57d5f8c51d0cce73253fde30aa2R246-R268